projects/plugins/crm/includes/ZeroBSCRM.DAL3.Obj.Segments.php

```diff
@@ -1435,7 +1435,24 @@ public function segmentConditionArgs($condition=array(),$conditionKeySuffix=''){
 
                     global $zbs,$wpdb,$ZBSCRM_t;
 
-                    switch ( $condition['type'] ){
+					$advanced_segments_active = defined( 'JPCRM_ADVANCED_SEGMENTS_ROOTFILE' );
+
+			// If the Advanced Segments plugin is active, we want to use the logic defined there instead.
+			if ( ! empty( $condition['type'] ) && $advanced_segments_active ) {
+				$filter_tag = $this->makeSlug( $condition['type'] ) . '_zbsSegmentArgumentBuild';
+				if ( substr( $condition['type'], 0, 5 ) === 'zbsc_' ) {
+					$filter_tag = $this->makeSlug( substr( $condition['type'], 5 ) ) . '_zbsSegmentArgumentBuild';
+				}
```

Comment:
I'd simplify this:

```suggestion
				$filter_tag = $this->makeSlug( preg_replace( '/^zbsc_/', '', $condition['type'] ) ) . '_zbsSegmentArgumentBuild';
```

Or perhaps more readable:

```suggestion
				// normalise type string
				$condition_type  = preg_replace( '/^zbsc_/', '', $condition['type'] );
				$filter_tag = $this->makeSlug( $condition_type ) . '_zbsSegmentArgumentBuild';
```